create article PTZ camera (#688)
airalab/robonomics-wiki: docs/ptz-camera.md

---
title: RTZ camera control in Home Assistant
contributors: [nakata5321]
---

In this article, you will set up RTZ camera in Home Assistant. 
To set up RTZ camera in Home assistant you will use ONVIF protocol. 

> To use it usually you have to set up local camera account, his process won't be covered in this article.

Reviewer: this process

Reviewer: and maybe highlight the warning that the process of setting up the local camera account is not going to be covered in the article

Requirements:
- RTZ camera
- Configured  camera **device** account
- IP address of camera
- Configured Home Assistant

## ONVIF integration

Let's start with installation  **ONVIF integration**. 

Go to "Devices & Services" in "Settings" and press "ADD INTEGRATION" button.
Type "ONVIF" and choose the integration. You will see the next window.

 <robo-wiki-picture src="home-assistant/onvifsetup.jpg" />

Press "Submit" button. It will try to automatic search of your camera. If it will be succeeding, choose your camera from list and 
fill empty fields. Or you have to fill all fields manually. You will see next window.

 <robo-wiki-picture src="home-assistant/onvifconfig.jpg" />

Fill the fields:
- Name - give a name to your camera
- Host - provide The IP address of your camera
- Port - 2020 (Usually port is 2020, but your camera provider may change port)
- Username - write a username of your camera **device** account
- Password - write a password of your camera **device** account

and press "Submit". Choose an Area for your camera and "Finish" installation.

## Add camera control to dashboard

> You already fully setup camera to Home Assistant. Now let's provide camera stream and control buttons to dashboard.

Reviewer: English. Please put Russian text to google translate and copy paste the result

---
title: RTZ camera control in Home Assistant
contributors: [nakata5321]
---

In this article, you will set up an RTZ camera in Home Assistant. 
To set up the RTZ camera in Home Assistant you will use the ONVIF protocol. 
To use it usually you have to set up a local camera account.

<robo-wiki-note type="warning">
the process of setting up the local camera account is not going to be covered in the article.
</robo-wiki-note>

Requirements:
- RTZ camera
- Configured  camera **device** account
- IP address of the camera
- Configured Home Assistant

## ONVIF integration

Let's start with the installation of **ONVIF integration**. 

Go to "Devices & Services" in "Settings" and press the "ADD INTEGRATION" button.
Type "ONVIF" and choose the integration. You will see the next window.

 <robo-wiki-picture src="home-assistant/onvifsetup.jpg" />

Press the "Submit" button. It will try to automatically search for your camera. If it will be succeeding, 
choose your camera from the list and fill in empty fields. 

> Or you have to fill in all fields manually. You will see the next window.

Reviewer 2: ```suggestion
Otherwise you have to fill in all the fields manually. You will see the following window.
```

 <robo-wiki-picture src="home-assistant/onvifconfig.jpg" />

> Fill in the fields:

Reviewer 2: ```suggestion
Fill in the gaps:
```

Reviewer 2: просто убрать повторения

- Name - give a name to your camera
- Host - provide The IP address of your camera
- Port - 2020 (Usually port is 2020, but your camera provider may change port)

> - Username - write a username of your camera **device** account

Reviewer 2: ```suggestion
- Username - write a username of your camera local account
```

> - Password - write a password for your camera **device** account

Reviewer 2: ```suggestion
- Password - write a password for your camera local account
```

> and press "Submit". Choose an Area for your camera and "Finish" installation.

Reviewer 2: ```suggestion
and press "Submit". Choose an Area for your camera and click on "Finish".
```

## Add camera control to the dashboard

> You already fully set up the camera to Home Assistant. 

Reviewer 2: ```suggestion
Now that you have fully set up the camera, you may add its stream and control buttons to the dashboard.
```

> Now let's provide a camera stream and control buttons to the dashboard.

Reviewer 2: ```suggestion

```

> Go to the dashboard and start creating a new card. Choose the "Picture Glance" card.

Reviewer 2: ```suggestion
Go to the dashboard and start with creating a new card. Choose the "Picture Glance" one.
```

 <robo-wiki-picture src="home-assistant/glance.jpg" />

> Change nest fields:

Reviewer 2: ```suggestion
Fill in the data:
```

> - Name - Write a place where you install the camera

Reviewer 2: ```suggestion
- Title - choose camera image title
```

> - Camera Entity - Choose a camera entity from the drop list

Reviewer 2: ```suggestion
- Camera Entity - choose a camera entity from the drop-down list
```

> - Camera View - choose "live" to get less time lag

Reviewer 2: ```suggestion
- Camera View - choose "live" to get less delay```

> Next, change to "Code Editor" mode by pressing the button at the bottom left side. You will see the next code:

Reviewer 2: ```suggestion
Next, switch to "Code Editor" mode by pressing the button at the bottom left side. You will see the following code:
```

```shell
camera_view: live
type: picture-glance
title: Kitchen
image: https://demo.home-assistant.io/stub_config/kitchen.png
entities: []
camera_image: camera.tapo_mainstream
```

> Instead of `entities: []` line insert next (The list of entities could be not empty. Then delete all lines with entities):

Reviewer 2: ```suggestion
Replace the content of `entities: []` according to the example below (`<YOUR_CAMERA_ENTITY>` is the same as the `camera_image` parameter):
```

      service: onvif.ptz
      service_data:
        entity_id: <YOUR_CAMERA_ENTITY>
        pan: LEFT
        speed: 1
        distance: 0.3
        move_mode: ContinuousMove
    name: Pan Left
    show_state: false
    icon: 'mdi:arrow-left'
    show_icon: true
  - entity: <YOUR_CAMERA_ENTITY>
    tap_action:
      action: call-service
      service: onvif.ptz
      service_data:
        entity_id: <YOUR_CAMERA_ENTITY>
        tilt: UP
        speed: 1
        distance: 0.3
        move_mode: ContinuousMove
    name: Tilt Up
    icon: 'mdi:arrow-up'
  - entity: <YOUR_CAMERA_ENTITY>
    tap_action:
      action: call-service
      service: onvif.ptz
      service_data:
        entity_id: <YOUR_CAMERA_ENTITY>
        tilt: DOWN
        speed: 1
        distance: 0.3
        move_mode: ContinuousMove
    name: Tilt Down
    icon: 'mdi:arrow-down'
  - entity: <YOUR_CAMERA_ENTITY>
    tap_action:
      action: call-service
      service: onvif.ptz
      service_data:
        entity_id: <YOUR_CAMERA_ENTITY>
        pan: RIGHT
        speed: 1
        distance: 0.3
        move_mode: ContinuousMove
    name: Pan Right
    icon: 'mdi:arrow-right'
    show_icon: true
```

</code-helper>

> Then copy the camera entity id from the "camera_image" key. (In the example it's `camera.tapo_mainstream`.) 

Reviewer 2: ```suggestion
```

> And insert it in place of `<YOUR_CAMERA_ENTITY>`. Save configuration.

Reviewer 2: ```suggestion
```

> That's all. Now you should see Card on the dashboard with control buttons.

Reviewer 2: ```suggestion
That's all. Now you should see the RTZ camera card on the dashboard along with control buttons.
```
